### PR TITLE
Fix ExtractList to support extraction from generic api.List{}

### PR DESCRIPTION
**Pull request by mfojtik**
@smarterclayton fyi.

### CODE REVIEW

File: pkg/runtime/helper_test.go [hunk 1]
```diff
@@ -63,6 +63,28 @@ func TestExtractList(t *testing.T) {
 	}
 }
 
+func TestExtractGenericList(t *testing.T) {
```
Comment (smarterclayton) [inline pkg/runtime/helper_test.go]: Can you add a few more test cases? Test []*runtime.Object, []api.Foo, []*api.Foo

Comment (mfojtik) [inline pkg/runtime/helper_test.go]: @smarterclayton I don't know what you mean with []*runtime.Object, []api.Foo is already tested in TestExtractList and []*api.Foo is what I'm testing

Comment (smarterclayton) [inline pkg/runtime/helper_test.go]: Test the variations of the arrays beyond just `[]interface` and `[]value` - test `[]pointer to interface` and `[]pointer to value`

Comment (mfojtik) [inline pkg/runtime/helper_test.go]: @smarterclayton I still don't get what do you want to test :-(

The ExtractList accepts an runtime.Object, so you can use only api.*List{} or you need to make up some fake resource like:

```
type testObj struct {
	Items []*runtime.Object
}
func (t testObj) IsAnAPIObject() {}
```

Now, how you can now populate the Items with values when the `runtime.Object` is already an interface and you basically saying that you want to have a list of pointers to interfaces there (how that can happen?)

Can you please make an example or elaborate more? :-(

**Comment by mfojtik** [comment]
@smarterclayton added more test cases, can you please double check them as I dunno if they exercise all combinations.

**Comment by smarterclayton** [comment]
Tests look fine, you need to gofmt

**Comment by mfojtik** [comment]
@smarterclayton done, weird for some reasons goimports and gofmt produce different results for me...

**Comment by smarterclayton** [comment]
LGTM

**Pull request merged**